File: redsheet-portal/src/app/database-models/ourbatna.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectRequirement,  } from './'
export class Ourbatna {
    constructor(ourbatna?: Ourbatna )  {
        let properties = [ "ourbatnaId", "ourbatnaValue", "projectRequirementId",];
        for (let property of properties)
            if (ourbatna && ourbatna[property])
                this[property] = ourbatna[property];
    }
 
	ourbatnaId : number =   0 ;
 
    @required()
    @maxLength(400)
	ourbatnaValue : string =   undefined;
 
    @range(0,2147483647)
	projectRequirementId : number =   undefined;
	projectRequirement : ProjectRequirement  ;


}

File: RedSheet.DbEntities/Models/Ourbatna.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("Ourbatnas", Schema = "dbo")]
	[Serializable]
	public partial class Ourbatna  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int OurbatnaId { get; set; } 
 
	[Required] 
	[MaxLength(400)]
	public string OurbatnaValue { get; set; } 
 
	[RelationshipTableAttribue("ProjectRequirements","dbo")]
	//Relationships
	public int ProjectRequirementId { get; set; } 
 
	[ForeignKey("ProjectRequirementId")]
	public ProjectRequirement ProjectRequirement { get; set; } 

	public Ourbatna()
	{
    }

    }
}

